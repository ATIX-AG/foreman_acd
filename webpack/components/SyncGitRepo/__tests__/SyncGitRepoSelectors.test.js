import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  selectScmType,
  selectPath,
  selectGitCommit,
  selectGitUrl,
} from '../SyncGitRepoSelectors';

import {
  syncGitRepoConfData_1,
} from '../__fixtures__/syncGitRepoConfData_1.fixtures';

const stateFactory = obj => ({
  foremanAcd: {
    syncGitRepoConf: obj,
  },
});

const fixtures = {
  'should return scmType from syncGitRepoConfData_1 fixtures': () =>
    selectScmType(stateFactory(syncGitRepoConfData_1)),
  'should return path from syncGitRepoConfData_1 fixtures': () =>
    selectPath(stateFactory(syncGitRepoConfData_1)),
  'should return gitCommit from syncGitRepoConfData_1 fixtures': () =>
    selectGitCommit(stateFactory(syncGitRepoConfData_1)),
  'should return gitUrl from syncGitRepoConfData_1 fixtures': () =>
    selectGitUrl(stateFactory(syncGitRepoConfData_1)),
};

describe('SyncGitRepoSelectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
