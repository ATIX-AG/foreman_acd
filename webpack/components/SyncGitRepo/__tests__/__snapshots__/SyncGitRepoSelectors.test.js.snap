// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncGitRepoSelectors should return gitCommit from syncGitRepoConfData_1 fixtures 1`] = `"master"`;

exports[`SyncGitRepoSelectors should return gitUrl from syncGitRepoConfData_1 fixtures 1`] = `"https://github.com/alpha.git"`;

exports[`SyncGitRepoSelectors should return path from syncGitRepoConfData_1 fixtures 1`] = `"https://example.com"`;

exports[`SyncGitRepoSelectors should return scmType from syncGitRepoConfData_1 fixtures 1`] = `
Object {
  "git": "git",
}
`;
